crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs

```diff
@@ -80,24 +82,146 @@ impl Visit for ReactRefreshUsageFinder {
   }
 }
 
-// __webpack_require__.$ReactRefreshRuntime$ is injected by the react-refresh additional entry
-// See https://github.com/web-infra-dev/rspack/pull/2714 why we have a promise here
-static RUNTIME_CODE: &str = r#"
-function $RefreshReg$(type, id) {
-  __webpack_modules__.$ReactRefreshRuntime$.register(type, __webpack_module__.id+ "_" + id);
+// $ReactRefreshRuntime$ is injected by provide
+fn create_react_refresh_runtime_stmts(unresolved_mark: Mark) -> Vec<Stmt> {
```

Comment:
This should be revert after fixed https://github.com/web-infra-dev/rspack/issues/3544